### PR TITLE
In-app content reporting

### LINKED ISSUE #5676: Content reports (dmca and others)
**LBRY needs to process content reports**
API is available at [reports.lbry.com](https://reports.lbry.com)

**Additional context**
link with specification of API
https://docs.google.com/spreadsheets/d/1CXUO0b6uU7hCNureMb0F-ruzrtqZyADNjk_JrKmaZzs/edit?usp=sharing

### PR DESCRIPTION
**Pull request by infinite-persistence**
## Issue
Closes #5676 [Content reports (dmca and others) ](https://github.com/lbryio/lbry-desktop/issues/5676)

## Screenshots
Here are some screenshots.  DMCA one includes more fields, but looks roughly the same.

<details>
  <summary>Click to expand</summary>
  
- image
- image
- image
- image
- image
- image

</details>

### CODE REVIEW

**Comment by tzarebczan** [comment]
Will this page accept a URL parameter with claim id? If not, let's please add that. This will be linked from other apps like IOS/Android/Desktop so the UI doesn't have to be re-created there.

**Comment by kauffj** [comment]
I missed the attached images before. Some feedback:

- Timestamp UI could be improved if it's easy
- Colons are almost always a sign that you can improve styling. Properly styled labels and values do not require colons.
- "Submit Report?" needs another title. Perhaps confirmation dialog could be styled after the confirm publish dialog?
- The green checkmark next to Report Submitted appears to introduce a new style. Either the dialog should close and a snackbar should be used, or the dialog can stay open and the feedback can be given by the colored bottom bar (I forget the name of this element).

File: ui/constants/report_content.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+export const PARTY_SELF = 'Myself';
+export const PARTY_GROUP = 'My company, organization, or client';
+
+export const RELATIONSHIP_FIELD_MIN_WIDTH = 10;
+export const COPYRIGHT_OWNER_MAX_LENGTH = 50;
+
+export const PARAMETERS = {
+  type: {
+    'Sexual content': {
+      CATEGORIES: [
```
Comment (seanyesmunt) [inline ui/constants/report_content.js:28]: This should be `[CATEGORIES]`

File: ui/redux/actions/reportContent.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import { COPYRIGHT_ISSUES, OTHER_LEGAL_ISSUES } from 'constants/report_content';
+
+type Dispatch = (action: any) => any;
+
+export const doReportContent = (category: string, params: string) => (dispatch: Dispatch) => {
+  dispatch({
+    type: ACTIONS.REPORT_CONTENT_STARTED,
+  });
+
+  let RICK_URL;
```
Comment (seanyesmunt) [inline ui/redux/actions/reportContent.js:12]: What does `RICK` mean?

File: ui/component/reportContent/view.jsx [hunk 3]
```diff
@@ -0,0 +1,727 @@
...
+
+      const seconds = window.player.currentTime();
+      const h = Math.floor(seconds / 3600);
+      const m = Math.floor((seconds % 3600) / 60);
+      const s = Math.floor((seconds % 3600) % 60);
+
+      const str = (n) => n.toLocaleString('en-US', { minimumIntegerDigits: 2, useGrouping: false });
+      updateInput('timestamp', str(h) + ':' + str(m) + ':' + str(s));
+    }
+  }, []);
```
Comment (seanyesmunt) [inline ui/component/reportContent/view.jsx:105]: `claimId` should be passed to this.

**Review by seanyesmunt — COMMENTED** [review]
A few small comments but this mostly looks good. This is a big improvement

File: ui/redux/actions/reportContent.js (hunk 2 above)
Comment (kauffj) [reply ui/redux/actions/reportContent.js:12]: `REPORT_CONTENT_URL`

Comment (seanyesmunt) [reply ui/redux/actions/reportContent.js:12]: nvm. This can probably be something like `REPORT_URL`

**Comment by infinite-persistence** [comment]
`--- Changes ---`
- Update:
    - `CATEGORIES` to `[CATEGORIES]`
    - `RICK_URL` to `REPORT_URL`
- Split out the effects, and correct the dependancies.
- Fix bug of not being able to proceed if reporting on Markdown because timestamp wasn't set.

**Comment by infinite-persistence** [comment]
Ping just in case ... anything else that needs changing?

**Comment by tzarebczan** [comment]
All good, I'll get this shipped tomorrow. Thanks again!

**Pull request merged**